Add revised `peak_local_max` to `skimage2` (#8039)
scikit-image/scikit-image: doc/source/user_guide/skimage2_migration.md

with `length`, `n_dim`, and `blob_size_fraction` containing values used with the old signature.
Other parameters -- including `boundary_mode` if you already set it explicitly -- can be left unchanged.

### `skimage.feature.peak_local_max`

This function is replaced by `skimage2.feature.peak_local_max` which now uses the Euclidean distance by default (`p_norm=2`).

> Every parameter but the first one – `image` – have been turned into keyword-only parameters.

Reviewer: ```suggestion
All parameters other than the first (`image`) are now keyword-only.
```

with `length`, `n_dim`, and `blob_size_fraction` containing values used with the old signature.
Other parameters -- including `boundary_mode` if you already set it explicitly -- can be left unchanged.

### `skimage.feature.peak_local_max`

This function is replaced by `skimage2.feature.peak_local_max` with new behavior:

- `p_norm` defaults to 2 (Euclidean distance), was `np.inf` (Chebyshev distance)
- `exclude_border` defaults to 0, was `True`
- `exclude_border` no longer accepts `False` and `True`, pass 0 instead of `False`, or `min_distance` instead of `True`

> - Parameters after `image` are keyword-only

Author: With every change – especially not allowing bool anymore – this gets harder to explain. See below.

I wonder if we should deprecate passing booleans with a conventional deprecation to simplify things into separate steps? 

Als not sure if it's really worth it at this stage. Stopping to accept bools is easy later one. Maybe we should be more conservative and restrict ourselves to the defaults in this case?

with `length`, `n_dim`, and `blob_size_fraction` containing values used with the old signature.
Other parameters -- including `boundary_mode` if you already set it explicitly -- can be left unchanged.

### `skimage.feature.peak_local_max`

This function is replaced by `skimage2.feature.peak_local_max` with new behavior:

- `p_norm` defaults to 2 (Euclidean distance), was `np.inf` (Chebyshev distance)
- `exclude_border` defaults to 0, was `True`
- `exclude_border` no longer accepts `False` and `True`, pass 0 instead of `False`, or `min_distance` instead of `True`
- Parameters after `image` are keyword-only

To keep the old behavior from `skimage` (v1.x) use:

```python
import skimage2 as ski2

ski2.feature.peak_local_max(
    ...,
    exclude_border=new_exclude_border,
    p_norm=old_p_norm,
)
```

where `old_p_norm` is the old default or given value.

> `new_exclude_border` should be derived from the old value according to the advice above.

Author: I've tried to improve this section in https://github.com/scikit-image/scikit-image/pull/8039/commits/eee2e507b5fef0af2cba8fe76be9406a9afcfa24. But I'm still not happy. I expect that a lot of users that relied on `min_distance=True` (explicitly or implicitly), will be confused. 

I wonder if it's useful raising a more useful `PendingSkimage2Change` warning that recommends a concrete updated signature.
